src/log.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Define logging helpers."""
+
+import functools
+
+
+def log_event_handler(logger):
+    """Log with the provided logger when a event handler method is executed.
+
+    Args:
+        logger: logger used to log events.
+
+    Returns:
+        Decorator wrapper.
+    """
+
+    def decorator(method):
+        """Log decorator wrapper.
+
+        Args:
+            method: method wrapped by the decorator.
+
+        Returns:
+            Decorated method.
+        """
+
+        @functools.wraps(method)
+        def decorated(self, event):
+            """Log decorator method.
+
+            Args:
+                event: The event triggered when the relation changes.
+
+            Returns:
+                Decorated method.
+            """
+            logger.info(f"* running {self.__class__.__name__}.{method.__name__}")
```

Comment:
The recommendation is not to use f-strings for logging, see https://juju.is/docs/sdk/styleguide#heading--templating